File: replace_sidebar.py

import os

filename = 'c:/Aldo/PetMate/petmate-landing-mvp/pages/sitter.tsx'
start_line = 669
end_line = 887

with open(filename, 'r', encoding='utf-8') as f:
    lines = f.readlines()

# Adjust for 0-based index
start_idx = start_line - 1
end_idx = end_line 

# Validate content?
print(f"Replacing lines {start_line} to {end_line}")
print("First line to remove:", lines[start_idx].strip())
print("Last line to remove:", lines[end_idx-1].strip())

new_content = [
    '                            <div className="bg-white rounded-2xl border border-slate-200 p-6 shadow-sm">\n',
    '                                <h2 className="text-xl font-bold">Sidebar Placeholder</h2>\n',
    '                            </div>\n'
]

final_lines = lines[:start_idx] + new_content + lines[end_idx:]

with open(filename, 'w', encoding='utf-8') as f:
    f.writelines(final_lines)

print("Replacement done.")

File: replace_sidebar.js

const fs = require('fs');

const filename = 'c:/Aldo/PetMate/petmate-landing-mvp/pages/sitter.tsx';
const startLine = 669;
const endLine = 887;

const content = fs.readFileSync(filename, 'utf-8');
const lines = content.split('\n');

// Adjust for 0-based index
const startIdx = startLine - 1;
const endIdx = endLine; // Slice end is exclusive, match endLine inclusive if removing

console.log(`Replacing lines ${startLine} to ${endLine}`);
if (lines.length > startIdx) {
    console.log("First line to remove:", lines[startIdx].trim());
}
if (lines.length >= endIdx) {
    console.log("Last line to remove:", lines[endIdx - 1].trim());
}

const newContent = [
    '                            <div className="bg-white rounded-2xl border border-slate-200 p-6 shadow-sm">',
    '                                <h2 className="text-xl font-bold">Sidebar Placeholder</h2>',
    '                            </div>'
];

const finalLines = [...lines.slice(0, startIdx), ...newContent, ...lines.slice(endIdx)];

fs.writeFileSync(filename, finalLines.join('\n'), 'utf-8');
console.log("Replacement done.");
